shell/machine-config/azure.vue

```diff
@@ -177,19 +199,192 @@ export default {
     },
   },
 
+  computed: {
+    locationOptionsInDropdown() {
+      const locationOptionsCopy = [...this.locationOptions];
+
+      return locationOptionsCopy.sort((a, b) => {
+        // Hopefully it's easier to find a region if the list is in
+        // alphabetical order.
+        if (a.name > b.name) {
+          return 1;
+        }
+        if (a.name < b.name) {
+          return -1;
+        }
+
+        return 0;
+      } )
+        .map((option) => {
+          return {
+            displayName: `${ option.displayName } (${ option.name })`,
+            name:        option.name,
+          };
+        });
+    },
+    vmsWithAcceleratedNetworking() {
+      return this.vmSizes.filter((vmData) => {
+        return vmData.AcceleratedNetworkingSupported;
+      });
+    },
+    vmsWithoutAcceleratedNetworking() {
+      return this.vmSizes.filter((vmData) => {
+        return !vmData.AcceleratedNetworkingSupported;
+      });
+    },
+    selectedVmSizeSupportsAN() {
+      const selectedSizeIsValid = !!this.vmsWithAcceleratedNetworking.find((vmData) => {
+        return this.value.size === vmData.Name;
+      });
+
+      return selectedSizeIsValid;
+    },
+    vmSizeAcceleratedNetworkingWarning() {
+      if (!this.selectedVmSizeSupportsAN) {
+        return this.t('cluster.machineConfig.azure.size.selectedSizeAcceleratedNetworkingWarning');
+      }
+
+      return '';
+    },
+    vmsWithAvailabilityZones() {
+      return this.vmSizes.filter((vmData) => {
+        return vmData.AvailabilityZones.length > 0;
+      });
+    },
+    vmSizeAvailabilityWarning() {
+      const selectedVmIsAvailableInSelectedRegion = this.vmSizes.filter((vmData) => {
+        return vmData.Name === this.value.size;
+      }).length > 0;
+
+      if (!selectedVmIsAvailableInSelectedRegion) {
+        return this.t('cluster.machineConfig.azure.size.availabilityWarning');
+      }
+
+      return '';
+    },
+    selectedVmSizeHasZones() {
+      const dataForSelectedSize = this.vmsWithAvailabilityZones.filter((vmData) => {
+        const { Name } = vmData;
+
+        return Name === this.value.size;
+      });
+
+      if (dataForSelectedSize.length > 0) {
+        return dataForSelectedSize[0].AvailabilityZones.length > 0;
+      }
+
+      return false;
+    },
+    vmAvailabilityZoneWarning() {
+      if (this.useAvailabilitySet) {
+        return '';
+      }
+      if (this.vmsWithAvailabilityZones.length === 0) {
+        return this.t('cluster.machineConfig.azure.size.regionDoesNotSupportAzs');
+      }
+
+      if (this.vmsWithAvailabilityZones > 0 && !this.selectedVmSizeHasZones) {
```

Comment:
I am never seeing the regionSupportsAzsButNotThisSize warning: it looks like this needs a `.length`

Reply:
I've made this change and the error is showing properly now:
<img width="955" alt="Screenshot 2022-12-16 at 2 53 06 PM" src="https://user-images.githubusercontent.com/20599230/208196196-743d6f9b-23fb-4d97-8f83-e2db40fece6c.png">